### PR TITLE
[MIG] 10 Product profile et example

### PR DESCRIPTION
**Pull request by EBII**

### CODE REVIEW

File: product_profile/models/product.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
+# coding: utf-8
+# © 2015 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+from odoo import models, fields, api, _
+from odoo.osv import orm
+from odoo.exceptions import Warning as UserError
```
Comment (Garamotte) [inline product_profile/models/product.py]: `from odoo.exceptions import UserError`

File: product_profile/models/product.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+             "(not synchronized with product.template fields)")
+    sequence = fields.Integer(
+        help="Defines the order of the entries of profile_id field\n"
+             "(not synchronized with product.template fields)")
+    explanation = fields.Text(
+        required=True,
+        oldname='description',
+        help="An explanation on the selected profile\n"
+             "(not synchronized with product.template fields)")
+    type = fields.Selection(
```
Comment (Garamotte) [inline product_profile/models/product.py:59]: Please don't use python's reserved names as field names.

File: product_profile/models/product.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+        selection='_get_types',
```
Comment (Garamotte) [inline product_profile/models/product.py]: New API provides a `selection_add` attribute, so you don't need to use a method to allow other modules to extend the selection list. You can directly put `selection=[('consu', 'Consumable'), ('service', 'Service')]` here instead.

File: product_profile/models/product.py [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        res = super(ProductProfile, self).write(new_vals)
+        if new_vals:
+            for rec in self:
+                products = self.env['product.product'].search(
+                    [('profile_id', '=', rec.id)])
+                if products:
+                    _logger.info(
+                        " >>> %s Products updating after updated '%s' "
+                        "product profile" % (len(products), rec.name))
+                    products.write({'profile_id': rec.id})
```
Comment (Garamotte) [inline product_profile/models/product.py]: I don't understand this.
You write `profile_id = rec.id` only on records that already have this value ? Or I miss something ?

File: product_profile/models/product.py [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+                    elm = etree.Element(
+                        'filter', **self._customize_profile_filters(my_filter))
+                    node[0].addprevious(elm)
+                res['arch'] = etree.tostring(doc, pretty_print=True)
+        return res
+
+    @api.model
+    def _get_profiles_to_filter(self):
+        """ Inherit if you want that some profiles doesn't have a filter """
+        return [(x.id, x.name) for x in self.env['product.profile'].search([])]
```
Comment (Garamotte) [inline product_profile/models/product.py:231]: This is exactly `return self.name_get()`, isn't it ?

File: product_profile/security/group.xml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data>
```
Comment (Garamotte) [inline product_profile/security/group.xml]: Replace `<openerp>` by `<odoo>` and remove `<data>`.

File: product_profile/views/config_view.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data>
```
Comment (Garamotte) [inline product_profile/views/config_view.xml]: Replace `<openerp>` by `<odoo>` and remove `<data>`.

File: product_profile/views/product_view.xml [hunk 8]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data>
```
Comment (Garamotte) [inline product_profile/views/product_view.xml]: Replace `<openerp>` by `<odoo>` and remove `<data>`.

File: product_profile_example/__openerp__.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
+# coding: utf-8
```
Comment (Garamotte) [inline product_profile_example/__openerp__.py:1]: Remove this file.

File: product_profile_example/product_product_demo.xml [hunk 10]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+<odoo>
+    <data noupdate="1">
```
Comment (Garamotte) [inline product_profile_example/product_product_demo.xml]: `<odoo noupdate="1">`

File: product_profile_example/profile.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+# © 2015 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
+
+
+class ProductProfile(models.Model):
+    _inherit = 'product.profile'
+
+    def _get_types(self):
```
Comment (Garamotte) [inline product_profile_example/profile.py]: Override the field using `selection_add=[('product', 'Stockable Product')]` when you remove this method in the other module.

**Review by Garamotte — CHANGES_REQUESTED** [review]
Could you add some unit tests on the `product_profile` module ?

File: product_profile/models/product.py (hunk 1 above)
Comment (EBII) [reply product_profile/models/product.py]: hello, 
the unit test exist in product_profile_example to test product_profile
please use it

File: product_profile/models/product.py (hunk 3 above)
Comment (EBII) [inline product_profile/models/product.py]: yes sure but is the odoo fields name : product.template.type so we use it.

File: product_profile/models/product.py (hunk 4 above)
Comment (EBII) [reply product_profile/models/product.py]: it is to force product update after  profile change
see next commit for update code

File: product_profile/models/product.py (hunk 5 above)
Comment (EBII) [reply product_profile/models/product.py:231]: but not the same model

Comment (Garamotte) [reply product_profile/models/product.py:231]: Oh sorry, I knew I missed something.
So : `return self.env['product.profile'].search([]).name_get()`.

File: product_profile/models/product.py (hunk 2 above)
Comment (EBII) [reply product_profile/models/product.py:59]: yes sure but is the odoo fields name : product.template.type so we use it.

File: product_profile/models/product.py (hunk 5 above)
Comment (EBII) [reply product_profile/models/product.py:231]: i remove it : not necessary

File: product_profile/models/product.py [hunk 12]
```diff
@@ -0,0 +1,269 @@
...
+                new_vals.pop(key)
+        # super call must be after check_useless_key_in_vals() call
+        # because we compare value before and after write
+        res = super(ProductProfile, self).write(new_vals)
+        if new_vals:
+            for rec in self:
+                products = self.env['product.product'].search(
+                    [('profile_id', '=', rec.id)])
+                data = products._get_profile_data({'profile_id': rec.id})
+                products.write(data)
```
Comment (bealdav) [inline product_profile/models/product.py:83]: missing logger before write https://github.com/OCA/product-attribute/commit/56995f899884de7b1d41ffe68140beca1bd38c75#diff-6564fbea77f6e37ef19b6a92abeb4a40R92

**Comment by EBII** [comment]
fail rebase remove PR to new one

**Pull request closed without merging**